### PR TITLE
Document infra attributes processor

### PR DESCRIPTION
**Pull request by liustanley**
### What does this PR do?

This PR adds a readme for the infra attributes processor.

### CODE REVIEW

File: comp/otelcol/otlp/components/processor/infraattributesprocessor/README.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
+# Infra Attributes Processor
+
+The infra attributes processor extracts [Kubernetes Datadog tags](https://docs.datadoghq.com/containers/kubernetes/tag/?tab=datadogoperator#out-of-the-box-tags) based on labels or annotations and assigns these tags as resource attributes on traces, metrics, and logs.
+
+When telemetry is exported from the otel-agent, these infra attributes will be converted into Kubernetes Datadog tags and used as metadata in [Container Monitoring](https://docs.datadoghq.com/containers/).
```
Comment (songy23) [inline comp/otelcol/otlp/components/processor/infraattributesprocessor/README.md:5]:
```suggestion
When telemetry is exported from the otel-agent, these infra attributes will be converted into Datadog tags and used as metadata in [Container Monitoring](https://docs.datadoghq.com/containers/).
```

File: comp/otelcol/otlp/components/processor/infraattributesprocessor/README.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
+# Infra Attributes Processor
+
+The infra attributes processor extracts [Kubernetes Datadog tags](https://docs.datadoghq.com/containers/kubernetes/tag/?tab=datadogoperator#out-of-the-box-tags) based on labels or annotations and assigns these tags as resource attributes on traces, metrics, and logs.
```
Comment (songy23) [inline comp/otelcol/otlp/components/processor/infraattributesprocessor/README.md:3]:
```suggestion
The infra attributes processor extracts [Kubernetes tags](https://docs.datadoghq.com/containers/kubernetes/tag/?tab=datadogoperator#out-of-the-box-tags) based on labels or annotations and assigns these tags as resource attributes on traces, metrics, and logs.
```

File: comp/otelcol/otlp/components/processor/infraattributesprocessor/README.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+## Configuration
+
+The infra attributes processor will be added automatically by the [converter component](../../../../converter/README.md). If you opted out of the converter, or you want to change the defaults, you are able to configure the processor as so:
+```
+processors:
+  infraattributes:
+    cardinality: 0
+```
+
+The infra attributes processor also needs to be included in the pipelines section in order to be enabled:
```
Comment (songy23) [inline comp/otelcol/otlp/components/processor/infraattributesprocessor/README.md:16]:
```suggestion
The infra attributes processor also needs to be included in the pipelines in order to take effect:
```

File: comp/otelcol/otlp/components/processor/infraattributesprocessor/README.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+| workloadmeta.KindContainerImageMetadata                              | `container.image.id`                        |
+| workloadmeta.KindECSTask                                             | `aws.ecs.task.arn`                          |
+| workloadmeta.KindKubernetesDeployment                                | `k8s.deployment.name`, `k8s.namespace.name` |
+| workloadmeta.KindKubernetesMetadata                                  | `k8s.namespace.name`, `k8s.node.name`       |
+| workloadmeta.KindKubernetesPod                                       | `k8s.pod.uid`                               |
+| workloadmeta.KindProcess                                             | `process.pid`                               |
+
+## List of Kubernetes Tags
+
+For the full list of Kubernetes Datadog Tags added by the infra attributes processor, see [comp/core/tagger/tags/tags.go](../../../../../../comp/core/tagger/tags/tags.go).
```
Comment (songy23) [inline comp/otelcol/otlp/components/processor/infraattributesprocessor/README.md:56]:
```suggestion
For the full list of Kubernetes Tags added by the infra attributes processor, see [comp/core/tagger/tags/tags.go](../../../../../../comp/core/tagger/tags/tags.go).
```

**Review by songy23 — APPROVED** [review]
Some nits, LGTM overall

File: comp/otelcol/otlp/components/processor/infraattributesprocessor/README.md [hunk 5]
```diff
@@ -0,0 +1,56 @@
+# Infra Attributes Processor
+
+The infra attributes processor extracts [Kubernetes tags](https://docs.datadoghq.com/containers/kubernetes/tag/?tab=datadogoperator#out-of-the-box-tags) based on labels or annotations and assigns these tags as resource attributes on traces, metrics, and logs.
+
+When telemetry is exported from the otel-agent, these infra attributes will be converted into Datadog tags and used as metadata in [Container Monitoring](https://docs.datadoghq.com/containers/).
+
+## Configuration
+
```
Comment (dineshg13) [inline comp/otelcol/otlp/components/processor/infraattributesprocessor/README.md:8]: Can you please add the requirement for SDK - viz contianer.id or host.name etc

File: comp/otelcol/otlp/components/processor/infraattributesprocessor/README.md [hunk 6]
```diff
@@ -0,0 +1,218 @@
...
+### Cardinality
+The cardinality option sets the [TagCardinality](../../../../../../comp/core/tagger/README.md#tagcardinality) in the Datadog Agent tagger component. Possible values for this option include:
+* `cardinality: 0` - **LowCardinality**: in the host count order of magnitude
+* `cardinality: 1` - **OrchestratorCardinality**: tags that change value for each pod or task
+* `cardinality: 2` - **HighCardinality**: typically tags that change value for each web request, user agent, container, etc.
+
+## Expected Attributes
+
+The infra attributes processor looks up the following resource attributes in order to extract Kubernetes Tags. These resource attributes can be set in your SDK or in your otel-agent collector configuration:
```
Comment (dineshg13) [inline comp/otelcol/otlp/components/processor/infraattributesprocessor/README.md:42]: looks up Tagger (Add link to tagger)

**Pull request merged**